packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -204,11 +207,16 @@ export class TerminalWidget extends BaseWidget implements StatefulWidget {
      * new terminal widget.
      * If id is provided attach to the terminal for this id.
      */
-    async start(id?: number): Promise<void> {
+    async start(id?: number): Promise<number> {
         this.terminalId = typeof id !== 'number' ? await this.createTerminal() : await this.attachTerminal(id);
         this.resizeTerminalProcess();
         this.connectTerminalProcess();
+        if (this.terminalId) {
```

Comment:
numbers always should be checked with `typeof this.terminalId === 'number'`, since `0` is a valid number, you can see how it breaks by refreshing the page with a single terminal

In this case there is an utility method: `IBaseTerminalServer.validateId(terminalId)`